pkg/es/config/config.go

```diff
@@ -44,48 +44,60 @@ import (
 	storageMetrics "github.com/jaegertracing/jaeger/storage/spanstore/metrics"
 )
 
+type TemplateOptions struct {
+	Priority    int `mapstructure:"priority"`
+	NumShards   int `mapstructure:"num_shards"`
+	NumReplicas int `mapstructure:"num_replicas"`
+}
+
+// IndexOptions describes the index format and rollover frequency
+type IndexOptions struct {
+	DateLayout        string          `mapstructure:"date_layout"`
+	RolloverFrequency string          `mapstructure:"rollover_frequency"`
+	TemplateOptions   TemplateOptions `mapstructure:"template"`
```

Comment:
Q: do you see much value in this being a substructure? It seems it would work just as well with `template_priority` `num_shards` `num_replicas` fields being defined here directly, no confusion, no extra nesting.

Reply:
My understanding is that there will be other parameters in the template in the future, which will be moved together to facilitate understanding.